core/src/main/scala/design/hamu/Board.scala

```diff
@@ -0,0 +1,63 @@
+package design.hamu
+
+//   Y
+//   ^
+// 2 | _ _ _
+// 1 | _ _ _
+// 0 | _ _ _
+//   +-------> X
+//     0 1 2
+
+class Board(var vecofvec: Vector[Vector[Char]]) {
+
+  // public function. call to init a new board with a new vecofvec, with one change in the given coordinate.
+  // returns same board with no changes if out of bounds
+  def move(targetX: Int, targetY: Int, targetPiece: Char): Board = {
+    try {
+      val oldCol = vecofvec(targetY)
+      val newCol = oldCol.updated(targetX, targetPiece)
+      val newBoard = vecofvec.updated(targetY, newCol)
+      new Board(newBoard)
+    } catch {
+      case x: IndexOutOfBoundsException => this
+    }
+
+  }
+
+  // public function. call to retrieve the piece in a given coordinate.
+  // returns tilde if out of bounds
+  def get(x: Int, y: Int): Char = {
+    try {
+      vecofvec(x)(y)
+    } catch {
+      case x: IndexOutOfBoundsException => '~'
+    }
+  }
+
+  def stringify(): String = {
+    var out = ""
```

Comment:
Var bad. val good.

Reply:
To avoid using var with for loop you can use `mkString` within `Vector` class which allows you to join a string using some delimiter.